src/Options/ParseOptions.hpp

```diff
@@ -273,22 +283,27 @@ template <typename OptionList, typename Group>
 void Parser<OptionList, Group>::parse_file(
     const std::string& file_name) noexcept {
   context_.append("In " + file_name);
+  errno = 0;
+  std::ifstream input(file_name);
+  if (not input) {
+    // There is no standard way to get an error message from an
+    // fstream, but this works on many implementations.
+    ERROR("Could not open " << file_name << ": " << strerror(errno));
+  }
+  input_source_.assign(std::istreambuf_iterator(input), {});
   try {
-    parse(YAML::LoadFile(file_name));
-  } catch (const YAML::BadFile& /*e*/) {
-    ERROR("Could not open the input file " << file_name);
-  } catch (const YAML::Exception& e) {
+    parse(YAML::Load(input_source_));
+  } catch (YAML::Exception& e) {
```

Comment:
Did you remove the `const` here for a particular reason? AFAICT the `parser_error` function takes its argument by const-ref

Reply:
Because I copied it from the next function where it was already missing.  I've fixed them both.  (Only change in the force push.)